---
title: "PCA and DAPC - pre vs post dam removal - steelhead and chinook"
author: "Kimberly Ledger"
date: "6/8/2022"
output: html_document
---

This code runs PCA/DAPCs on neutral loci only and all loci for steelhead and chinook salmon pre and post dam removal 

load libraries
```{r, message = FALSE}
library(vcfR) #this package is used to visualize and manipulate VCF files
library(adegenet) #this package is used for analysis of genetic/genomic data 
library(tidyverse) # data manipulation
library(forcats)
library(ggplot2)
```

# Part 1: Prepare the data for O. mykiss (steelhead and rainbow trout) 

## read in the steelhead (Oncorhynchus mykiss) metadata
```{r}
onmy_metadata <- read.csv("~/Desktop/LG_Proj4/Elwha_datafiles/Elwha_Steelhead_Formatted_kjl.csv") %>%
  dplyr::mutate(Location = fct_relevel(Location, "BD", "SBLR", "ID", "AD"))
head(onmy_metadata)
nrow(onmy_metadata)
```

## read in the O. mykiss VCF file - this file does not include cutthroat trout hyrids
```{r}
onmy_vcf <- read.vcfR("~/Desktop/LG_Proj4/Elwha_datafiles/Elwha_GTSeq_Sans_CCT.vcf")
onmy_vcf
```


## extract and filter genotypes for missing data  

this starts with extracting the genotypes from the vcfR object 
```{r}
geno <- extract.gt(onmy_vcf)
dim(geno)

head(geno[,1:10])
```

check vcf file for missing data 
```{r}
myMiss_loci <- apply(geno, MARGIN = 1, function(x){ sum(is.na(x)) })  #MARGIN = 1 refers to rows (loci)
myMiss_loci <- myMiss_loci/ncol(geno)

barplot(myMiss_loci, main = "Loci")
title(ylab = "Missingness (%)")

myMiss_sample <- apply(geno, MARGIN = 2, function(x){ sum(is.na(x)) })  #MARGIN = 2 refers to columns (samples)
myMiss_sample <- myMiss_sample/nrow(geno)

barplot(myMiss_sample, main = "Samples")
title(ylab = "Missingness (%)")

hist(myMiss_sample, main = "Samples with missing data")
```

remove samples with >20% missing genotype data 
```{r}
onmy_vcf@gt <- onmy_vcf@gt[, c(TRUE, myMiss_sample < 0.2)]
onmy_vcf

geno_filtered <- extract.gt(onmy_vcf)
dim(geno_filtered)
```



## read in loci metadata
```{r}
onmy_loci_meta <- read.csv("~/Desktop/LG_Proj4/Elwha_datafiles/Steelhead_Locus_Key_kjl.csv")
```

## read in SNP coordinate data 
```{r}
onmy_snp_coord <- read.csv("~/Desktop/LG_Proj4/Elwha_datafiles/SNP_Coordinates_CRITFC.csv")
```

join loci metadata and SNP coords 
```{r}
onmy_locus_join <- onmy_snp_coord %>%
  left_join(onmy_loci_meta, by = "Locus")
```

## join loci metadata to extracted genotypes 
```{r}
onmy_snp <- rownames(geno_filtered)

onmy_gt_df <- as.data.frame(geno_filtered)
onmy_gt_df$SNP <- onmy_snp

onmy_gt_meta <- onmy_gt_df %>%
  left_join(onmy_locus_join, by = "SNP")
```


loci NC_035104.1_11667915 is duplicated so we have to remove one of the rows
```{r}
onmy_gt_meta <- onmy_gt_meta[-304,]
```



## filter data by neutral and adaptive loci 
```{r}
onmy_neutral <- onmy_gt_meta %>%
  filter(Status == "Neutral")
dim(onmy_neutral)
onmy_adaptive <- onmy_gt_meta %>%
  filter(Status == "Adaptive")
dim(onmy_adaptive)
```

now only 329 loci - check if we have info on whether the 5 missing loci are neutral or adaptive

## seperate filtered neutral and adaptive genotype data into a pre-dam and post-dam datasets 
```{r}
pre_id <- onmy_metadata$Sample_ID[which(onmy_metadata$Time=="Pre")]
post_id <- onmy_metadata$Sample_ID[which(onmy_metadata$Time=="Post")]

rownames(onmy_neutral) <- onmy_neutral$SNP
neutral_tG <- t(onmy_neutral[,c(1:1324)])

rownames(onmy_adaptive) <- onmy_adaptive$SNP
adaptive_tG <- t(onmy_adaptive[,c(1:1324)])

geno_neutral_pre <- neutral_tG[rownames(neutral_tG) %in% pre_id,]
dim(geno_neutral_pre)
geno_neutral_post <- neutral_tG[rownames(neutral_tG) %in% post_id,]
dim(geno_neutral_post)
geno_adaptive_pre <- adaptive_tG[rownames(adaptive_tG) %in% pre_id,]
dim(geno_adaptive_pre)
geno_adaptive_post <- adaptive_tG[rownames(adaptive_tG) %in% post_id,]
dim(geno_adaptive_post)
```


# create genind objects 

### neutral loci pre-dam removal
```{r}
colmf <- gsub("\\.", "_", colnames(geno_neutral_pre))
colnames(geno_neutral_pre) <- colmf

n_pre_loci <- colnames(geno_neutral_pre)
n_pre_ind <- rownames(geno_neutral_pre)

ind_df <- data.frame(n_pre_ind)
colnames(ind_df) <- "Sample_ID"
onmy_n_pre_meta <- ind_df %>%
  left_join(onmy_metadata, by = "Sample_ID")

n_pre_location <- onmy_n_pre_meta$Location

genind_n_pre <- df2genind(geno_neutral_pre,
                         sep="/",
                         ind.names=n_pre_ind,
                         loc.names=n_pre_loci, 
                         pop = n_pre_location,     
                         ploidy = 2)
genind_n_pre
```

### neutral loci post-dam removal
```{r}
colmf <- gsub("\\.", "_", colnames(geno_neutral_post))
colnames(geno_neutral_post) <- colmf

n_post_loci <- colnames(geno_neutral_post)
n_post_ind <- rownames(geno_neutral_post)

ind_df <- data.frame(n_post_ind)
colnames(ind_df) <- "Sample_ID"
onmy_n_post_meta <- ind_df %>%
  left_join(onmy_metadata, by = "Sample_ID")

n_post_location <- onmy_n_post_meta$Location

genind_n_post <- df2genind(geno_neutral_post,
                         sep="/",
                         ind.names=n_post_ind,
                         loc.names=n_post_loci, 
                         pop = n_post_location,     
                         ploidy = 2)
genind_n_post
```

### adaptive loci pre-dam removal
```{r}
colmf <- gsub("\\.", "_", colnames(geno_adaptive_pre))
colnames(geno_adaptive_pre) <- colmf

a_pre_loci <- colnames(geno_adaptive_pre)
a_pre_ind <- rownames(geno_adaptive_pre)

ind_df <- data.frame(a_pre_ind)
colnames(ind_df) <- "Sample_ID"
onmy_a_pre_meta <- ind_df %>%
  left_join(onmy_metadata, by = "Sample_ID")

a_pre_location <- onmy_a_pre_meta$Location

genind_a_pre <- df2genind(geno_adaptive_pre,
                         sep="/",
                         ind.names=a_pre_ind,
                         loc.names=a_pre_loci, 
                         pop = a_pre_location,     
                         ploidy = 2)
genind_a_pre
```

### adaptive loci post-dam removal
```{r}
colmf <- gsub("\\.", "_", colnames(geno_adaptive_post))
colnames(geno_adaptive_post) <- colmf

a_post_loci <- colnames(geno_adaptive_post)
a_post_ind <- rownames(geno_adaptive_post)

ind_df <- data.frame(a_post_ind)
colnames(ind_df) <- "Sample_ID"
onmy_a_post_meta <- ind_df %>%
  left_join(onmy_metadata, by = "Sample_ID")

a_post_location <- onmy_a_post_meta$Location

genind_a_post <- df2genind(geno_adaptive_post,
                         sep="/",
                         ind.names=a_post_ind,
                         loc.names=a_post_loci, 
                         pop = a_post_location,     
                         ploidy = 2)
genind_a_post
```


##convet genind to genlight objects 

```{r, message=FALSE}
#install_github("green-striped-gecko/dartR")
library(dartR)

genlit_n_pre <- gi2gl(genind_n_pre)
genlit_n_post <- gi2gl(genind_n_post)
genlit_a_pre <- gi2gl(genind_a_pre)
genlit_a_post <- gi2gl(genind_a_post)
```


## PCA using genlight object

set color palette
```{r}
library(viridisLite)
mycol = viridisLite::viridis(4, alpha = 0.6)
```

run and plot PCA
```{r}
pca_n_pre <- glPca(genlit_n_pre, center = T, scale = F, nf = 50)
s.class(pca_n_pre$scores, pop(genlit_n_pre),
        xax=1, yax=2, col=mycol,
        axesel=FALSE, cstar=0, cpoint=3)
```

run and plot PCA
```{r}
pca_n_post <- glPca(genlit_n_post, center = T, scale = F, nf = 50)
s.class(pca_n_post$scores, pop(genlit_n_post),
        xax=1, yax=2, col=mycol,
        axesel=FALSE, cstar=0, cpoint=3)
```


run and plot PCA
```{r}
pca_a_pre <- glPca(genlit_a_pre, center = T, scale = F, nf = 50)
s.class(pca_a_pre$scores, pop(genlit_a_pre),
        xax=1, yax=2, col=mycol,
        axesel=FALSE, cstar=0, cpoint=3)
```

run and plot PCA
```{r}
pca_a_post <- glPca(genlit_a_post, center = T, scale = F, nf = 50)
s.class(pca_a_post$scores, pop(genlit_a_post),
        xax=1, yax=2, col=mycol,
        axesel=FALSE, cstar=0, cpoint=3)
```



# Part 5: DAPC

i am following this tutorial: https://adegenet.r-forge.r-project.org/files/tutorial-dapc.pdf

find clusters
```{r}
grp_n_pre <- find.clusters(genlit_n_pre, max.n.clust=40, n.pca=100)
```

check Kstat
```{r}
grp_n_pre$Kstat
```

will use K=3 and K=4 BIC
```{r}
grp_n_pre_3 <- find.clusters(genlit_n_pre, max.n.clust=40, n.clust = 3)
grp_n_pre_4 <- find.clusters(genlit_n_pre, max.n.clust=40, n.clust = 4)
```

run dapc
```{r}
dapc_n_pre_3 <- dapc(genlit_n_pre, grp_n_pre_3$grp, n.pca = 40, n.da = 10) #keep first 40 PCs; and 10 n.da
dapc_n_pre_3
dapc_n_pre_4 <- dapc(genlit_n_pre, grp_n_pre_4$grp, n.pca = 40, n.da = 10) #keep first 40 PCs; and 10 n.da
dapc_n_pre_4 
```

plot DAPC using ggplot
```{r}
dapc_n_pre_3_scores <- as.data.frame(dapc_n_pre_3$tab)
dapc_n_pre_3_scores$Assigned_Pop <- dapc_n_pre_3$grp
dapc_n_pre_3_scores$Sampling_Location <- onmy_n_pre_meta$Location

set.seed(9)
dapc_n_pre_3 <- ggplot(dapc_n_pre_3_scores, aes(x=PC1, y=PC2, shape = Assigned_Pop)) + 
  geom_point(size=2, aes(colour=Sampling_Location )) + 
  stat_ellipse(level = 0.95, size = 1) + 
  scale_color_manual(values = viridis(4)) + 
  geom_hline(yintercept = 0) + 
  geom_vline(xintercept = 0) + 
  theme_bw()
dapc_n_pre_3
```


### neutral post dam removal 

find clusters
```{r}
grp_n_post <- find.clusters(genlit_n_post, max.n.clust=40, n.pca=100)
```

check Kstat
```{r}
grp_n_post$Kstat
```

will use K= 3 and K=5
```{r}
grp_n_post_3 <- find.clusters(genlit_n_post, max.n.clust=40, n.clust = 3)
grp_n_post_5 <- find.clusters(genlit_n_post, max.n.clust=40, n.clust = 5)
```

run dapc
```{r}
dapc_n_post_3 <- dapc(genlit_n_post, grp_n_post_3$grp, n.pca = 40, n.da = 10) #keep first 40 PCs; and 10 n.da
dapc_n_post_3
```

plot DAPC using ggplot
```{r}
dapc_n_post_3_scores <- as.data.frame(dapc_n_post_3$tab)
dapc_n_post_3_scores$Assigned_Pop <- dapc_n_post_3$grp
dapc_n_post_3_scores$Sampling_Location <- onmy_n_post_meta$Location

set.seed(9)
dapc_n_post_3 <- ggplot(dapc_n_post_3_scores, aes(x=PC1, y=PC2, shape = Assigned_Pop)) + 
  geom_point(size=2, aes(colour=Sampling_Location )) + 
  stat_ellipse(level = 0.95, size = 1) + 
  scale_color_manual(values = viridis(4)) + 
  geom_hline(yintercept = 0) + 
  geom_vline(xintercept = 0) + 
  theme_bw()
dapc_n_post_3
```

### adaptive pre-dam removal

find clusters
```{r}
grp_a_pre <- find.clusters(genlit_a_pre, max.n.clust=40, n.pca=100)
```

check Kstat
```{r}
grp_a_pre$Kstat
```

will use K=4 and K=5 BIC
```{r}
grp_a_pre_4 <- find.clusters(genlit_a_pre, max.n.clust=40, n.clust = 4)
grp_a_pre_5 <- find.clusters(genlit_a_pre, max.n.clust=40, n.clust = 5)
```

run dapc
```{r}
dapc_a_pre_4 <- dapc(genlit_a_pre, grp_a_pre_4$grp, n.pca = 40, n.da = 10) #keep first 40 PCs; and 10 n.da
dapc_a_pre_4
dapc_a_pre_5 <- dapc(genlit_a_pre, grp_a_pre_5$grp, n.pca = 40, n.da = 10) #keep first 40 PCs; and 10 n.da
dapc_a_pre_5 
```


plot DAPC using ggplot
```{r}
dapc_a_pre_4_scores <- as.data.frame(dapc_a_pre_4$tab)
dapc_a_pre_4_scores$Assigned_Pop <- dapc_a_pre_4$grp
dapc_a_pre_4_scores$Sampling_Location <- onmy_a_pre_meta$Location

set.seed(9)
dapc_a_pre_4 <- ggplot(dapc_a_pre_4_scores, aes(x=PC1, y=PC2, shape = Assigned_Pop)) + 
  geom_point(size=2, aes(colour=Sampling_Location )) + 
  stat_ellipse(level = 0.95, size = 1) + 
  scale_color_manual(values = viridis(4)) + 
  geom_hline(yintercept = 0) + 
  geom_vline(xintercept = 0) + 
  theme_bw()
dapc_a_pre_4
```



### adaptive post-dam removal

find clusters
```{r}
grp_a_post <- find.clusters(genlit_a_post, max.n.clust=40, n.pca=100)
```

check Kstat
```{r}
grp_a_post$Kstat
```

will use K=4 and K=5 BIC
```{r}
grp_a_post_4 <- find.clusters(genlit_a_post, max.n.clust=40, n.clust = 4)
grp_a_post_5 <- find.clusters(genlit_a_post, max.n.clust=40, n.clust = 5)
```

run dapc
```{r}
dapc_a_post_4 <- dapc(genlit_a_post, grp_a_post_4$grp, n.pca = 40, n.da = 10) #keep first 40 PCs; and 10 n.da
dapc_a_post_4
dapc_a_post_5 <- dapc(genlit_a_post, grp_a_post_5$grp, n.pca = 40, n.da = 10) #keep first 40 PCs; and 10 n.da
dapc_a_post_5 
```


plot DAPC using ggplot
```{r}
dapc_a_post_4_scores <- as.data.frame(dapc_a_post_4$tab)
dapc_a_post_4_scores$Assigned_Pop <- dapc_a_post_4$grp
dapc_a_post_4_scores$Sampling_Location <- onmy_a_post_meta$Location

set.seed(9)
dapc_a_post_4 <- ggplot(dapc_a_post_4_scores, aes(x=PC1, y=PC2, shape = Assigned_Pop)) + 
  geom_point(size=2, aes(colour=Sampling_Location )) + 
  stat_ellipse(level = 0.95, size = 1) + 
  scale_color_manual(values = viridis(4)) + 
  geom_hline(yintercept = 0) + 
  geom_vline(xintercept = 0) + 
  theme_bw()
dapc_a_post_4
```


# now try working with the Chinook salmon data

## read in chinook data
```{r}
chin_meta <- read.csv("~/Desktop/LG_Proj4/Elwha_datafiles/Elwha_Chinook_Formatted_wDownstreamLatLongrkm_kjl.csv")
```

## read in the chinook VCF file - this file does not include cutthroat trout hyrids
```{r}
chin_vcf <- read.vcfR("~/Desktop/LG_Proj4/Elwha_datafiles/ElwhaGenos.vcf")
chin_vcf
```


## extract and filter genotypes for missing data  

this starts with extracting the genotypes from the vcfR object 
```{r}
c_geno <- extract.gt(chin_vcf)
dim(c_geno)

head(c_geno[,1:10])
```

check vcf file for missing data 
```{r}
myMiss_loci <- apply(c_geno, MARGIN = 1, function(x){ sum(is.na(x)) })  #MARGIN = 1 refers to rows (loci)
myMiss_loci <- myMiss_loci/ncol(c_geno)

barplot(myMiss_loci, main = "Loci")
title(ylab = "Missingness (%)")

myMiss_sample <- apply(c_geno, MARGIN = 2, function(x){ sum(is.na(x)) })  #MARGIN = 2 refers to columns (samples)
myMiss_sample <- myMiss_sample/nrow(c_geno)

barplot(myMiss_sample, main = "Samples")
title(ylab = "Missingness (%)")

hist(myMiss_sample, main = "Samples with missing data")
```

remove samples with >20% missing genotype data 
```{r}
chin_vcf@gt <- chin_vcf@gt[, c(TRUE, myMiss_sample < 0.2)]
chin_vcf

c_geno_filtered <- extract.gt(chin_vcf)
dim(c_geno_filtered)
```



## read in loci metadata

```{r}
chin_loci_key <- read.csv("~/Desktop/LG_Proj4/Elwha_datafiles/Chinook_Locus_Key.csv")
chin_loci_key <- chin_loci_key[,c(1:9)]
```

## join loci metadata to extracted genotypes 
```{r}
chin_snp <- rownames(c_geno_filtered)

chin_gt_df <- as.data.frame(c_geno_filtered)
chin_gt_df$Locus_Name <- chin_snp

chin_gt_meta <- chin_gt_df %>%
  left_join(chin_loci_key, by = "Locus_Name")
```


## filter data by neutral and adaptive loci 
```{r}
chin_neutral <- chin_gt_meta %>%
  filter(Status == "Neutral")
dim(chin_neutral)
chin_adaptive <- chin_gt_meta %>%
  filter(Status == "Adaptive")
dim(chin_adaptive)
```


## seperate filtered neutral and adaptive genotype data into a pre-dam (190 ind) and post-dam (164 ind) datasets 
```{r}
pre_id <- chin_meta$Sample_ID[which(chin_meta$Time=="Pre")]
post_id <- chin_meta$Sample_ID[which(chin_meta$Time=="Post")]

rownames(chin_neutral) <- chin_neutral$Locus_Name
neutral_tG <- t(chin_neutral[,c(1:359)])  

rownames(chin_adaptive) <- chin_adaptive$Locus_Name
adaptive_tG <- t(chin_adaptive[,c(1:359)])

geno_neutral_pre <- neutral_tG[rownames(neutral_tG) %in% pre_id,]
dim(geno_neutral_pre)
geno_neutral_post <- neutral_tG[rownames(neutral_tG) %in% post_id,]
dim(geno_neutral_post)
geno_adaptive_pre <- adaptive_tG[rownames(adaptive_tG) %in% pre_id,]
dim(geno_adaptive_pre)
geno_adaptive_post <- adaptive_tG[rownames(adaptive_tG) %in% post_id,]
dim(geno_adaptive_post)
```








## join data frames 
```{r}
chin_geno$Sample_ID <- rownames(chin_geno)

m <- chin_geno %>%
  left_join(chin_meta, by = "Sample_ID")
```

# create genind objects for chinook 

### neutral loci pre-dam removal
```{r}
colmf <- gsub("\\.", "_", colnames(geno_neutral_pre))
colnames(geno_neutral_pre) <- colmf

n_pre_loci <- colnames(geno_neutral_pre)
n_pre_ind <- rownames(geno_neutral_pre)

ind_df <- data.frame(n_pre_ind)
colnames(ind_df) <- "Sample_ID"
chin_n_pre_meta <- ind_df %>%
  left_join(chin_meta, by = "Sample_ID")

n_pre_location <- chin_n_pre_meta$Location

chin_genind_n_pre <- df2genind(geno_neutral_pre,
                         sep="/",
                         ind.names=n_pre_ind,
                         loc.names=n_pre_loci, 
                         pop = n_pre_location,     
                         ploidy = 2)
chin_genind_n_pre
```

### neutral loci post-dam removal
```{r}
colmf <- gsub("\\.", "_", colnames(geno_neutral_post))
colnames(geno_neutral_post) <- colmf

n_post_loci <- colnames(geno_neutral_post)
n_post_ind <- rownames(geno_neutral_post)

ind_df <- data.frame(n_post_ind)
colnames(ind_df) <- "Sample_ID"
chin_n_post_meta <- ind_df %>%
  left_join(chin_meta, by = "Sample_ID")

n_post_location <- chin_n_post_meta$Location

chin_genind_n_post <- df2genind(geno_neutral_post,
                         sep="/",
                         ind.names=n_post_ind,
                         loc.names=n_post_loci, 
                         pop = n_post_location,     
                         ploidy = 2)
chin_genind_n_post
```

### adaptive loci pre-dam removal
```{r}
colmf <- gsub("\\.", "_", colnames(geno_adaptive_pre))
colnames(geno_adaptive_pre) <- colmf

a_pre_loci <- colnames(geno_adaptive_pre)
a_pre_ind <- rownames(geno_adaptive_pre)

ind_df <- data.frame(a_pre_ind)
colnames(ind_df) <- "Sample_ID"
chin_a_pre_meta <- ind_df %>%
  left_join(chin_meta, by = "Sample_ID")

a_pre_location <- chin_a_pre_meta$Location

chin_genind_a_pre <- df2genind(geno_adaptive_pre,
                         sep="/",
                         ind.names=a_pre_ind,
                         loc.names=a_pre_loci, 
                         pop = a_pre_location,     
                         ploidy = 2)
chin_genind_a_pre
```

### adaptive loci post-dam removal
```{r}
colmf <- gsub("\\.", "_", colnames(geno_adaptive_post))
colnames(geno_adaptive_post) <- colmf

a_post_loci <- colnames(geno_adaptive_post)
a_post_ind <- rownames(geno_adaptive_post)

ind_df <- data.frame(a_post_ind)
colnames(ind_df) <- "Sample_ID"
chin_a_post_meta <- ind_df %>%
  left_join(chin_meta, by = "Sample_ID")

a_post_location <- chin_a_post_meta$Location

chin_genind_a_post <- df2genind(geno_adaptive_post,
                         sep="/",
                         ind.names=a_post_ind,
                         loc.names=a_post_loci, 
                         pop = a_post_location,     
                         ploidy = 2)
chin_genind_a_post
```


##convet genind to genlight objects 

```{r, message=FALSE}
#install_github("green-striped-gecko/dartR")
library(dartR)

chin_genlit_n_pre <- gi2gl(chin_genind_n_pre)
chin_genlit_n_post <- gi2gl(chin_genind_n_post)
chin_genlit_a_pre <- gi2gl(chin_genind_a_pre)
chin_genlit_a_post <- gi2gl(chin_genind_a_post)
```


## PCA using genlight objects 

set color palette
```{r}
library(viridisLite)
mycol = viridisLite::viridis(4, alpha = 0.6)
```

run and plot PCA
```{r}
chin_pca_n_pre <- glPca(chin_genlit_n_pre, center = T, scale = F, nf = 50)
s.class(chin_pca_n_pre$scores, pop(chin_genlit_n_pre),
        xax=1, yax=2, col=mycol[c(1,3,4)],
        axesel=FALSE, cstar=0, cpoint=3)
```

make nicer looking pca plot 
```{r}
pc_chin_n_pre <- data.frame(chin_pca_n_pre$scores)
pc_chin_n_pre$Location <- chin_n_pre_meta$Location
pc_chin_n_pre <- pc_chin_n_pre %>%  
  dplyr::mutate(Location = fct_relevel(Location, "BD", "ID", "AD"))

plot_chin_n_pre <- ggplot(data = pc_chin_n_pre, aes(x=PC1, y = PC2, col = Location)) + 
  geom_point(aes(col = Location), size = 3, alpha=0.8, show.legend = F) + 
  labs(color = "Location")+
  scale_color_manual(values=mycol[c(1,3,4)]) + 
  #stat_ellipse(level = 0.95, size = 1) + 
  geom_hline(yintercept = 0) + 
  geom_vline(xintercept = 0) + 
  theme_bw()
plot_chin_n_pre 
```


run and plot PCA
```{r}
chin_pca_n_post <- glPca(chin_genlit_n_post, center = T, scale = F, nf = 50)
s.class(chin_pca_n_post$scores, pop(chin_genlit_n_post),
        xax=1, yax=2, col=mycol[c(1,3,4)],
        axesel=FALSE, cstar=0, cpoint=3)
```

make nicer looking pca plot 
```{r}
pc_chin_n_post <- data.frame(chin_pca_n_post$scores)
pc_chin_n_post$Location <- chin_n_post_meta$Location
pc_chin_n_post <- pc_chin_n_post %>%  
  dplyr::mutate(Location = fct_relevel(Location, "BD", "ID", "AD"))

plot_chin_n_post <- ggplot(data = pc_chin_n_post, aes(x=PC1, y = PC2, col = Location)) + 
  geom_point(aes(col = Location), size = 3, alpha=0.8, show.legend = F) + 
  labs(color = "Location")+
  scale_color_manual(values=mycol[c(1,3,4)]) + 
  #stat_ellipse(level = 0.95, size = 1) + 
  geom_hline(yintercept = 0) + 
  geom_vline(xintercept = 0) + 
  theme_bw()
plot_chin_n_post 
```



run and plot PCA
```{r}
chin_pca_a_pre <- glPca(chin_genlit_a_pre, center = T, scale = F, nf = 50)
s.class(chin_pca_a_pre$scores, pop(chin_genlit_a_pre),
        xax=1, yax=2, col=mycol[c(1,3,4)],
        axesel=FALSE, cstar=0, cpoint=3)
```

make nicer looking pca plot 
```{r}
pc_chin_a_pre <- data.frame(chin_pca_a_pre$scores)
pc_chin_a_pre$Location <- chin_a_pre_meta$Location
pc_chin_a_pre <- pc_chin_a_pre %>%  
  dplyr::mutate(Location = fct_relevel(Location, "BD", "ID", "AD"))

plot_chin_a_pre <- ggplot(data = pc_chin_a_pre, aes(x=PC1, y = PC2, col = Location)) + 
  geom_point(aes(col = Location), size = 3, alpha=0.8, show.legend = F) + 
  labs(color = "Location")+
  scale_color_manual(values=mycol[c(1,3,4)]) + 
  #stat_ellipse(level = 0.95, size = 1) + 
  geom_hline(yintercept = 0) + 
  geom_vline(xintercept = 0) + 
  theme_bw()
plot_chin_a_pre 
```

run and plot PCA
```{r}
chin_pca_a_post <- glPca(chin_genlit_a_post, center = T, scale = F, nf = 50)
s.class(chin_pca_a_post$scores, pop(chin_genlit_a_post),
        xax=1, yax=2, col=mycol[c(1,3,4)],
        axesel=FALSE, cstar=0, cpoint=3)
```

make nicer looking pca plot 
```{r}
pc_chin_a_post <- data.frame(chin_pca_a_post$scores)
pc_chin_a_post$Location <- chin_a_post_meta$Location
pc_chin_a_post <- pc_chin_a_post %>%  
  dplyr::mutate(Location = fct_relevel(Location, "BD", "ID", "AD"))

plot_chin_a_post <- ggplot(data = pc_chin_a_post, aes(x=PC1, y = PC2, col = Location)) + 
  geom_point(aes(col = Location), size = 3, alpha=0.8, show.legend = F) + 
  labs(color = "Location")+
  scale_color_manual(values=mycol[c(1,3,4)]) + 
  #stat_ellipse(level = 0.95, size = 1) + 
  geom_hline(yintercept = 0) + 
  geom_vline(xintercept = 0) + 
  theme_bw()
plot_chin_a_post 
```

why are there pre-dam removal samples from ID and AD? 


plot all together 
```{r}
library(cowplot)

neutral <- plot_grid(plot_chin_n_pre, plot_chin_n_post, labels = "AUTO", ncol = 2)
neutral

adaptive <- plot_grid(plot_chin_a_pre, plot_chin_a_post, labels = "AUTO", ncol = 2)
adaptive
```




# Part 5: DAPC

### pre-dam removal neutral loci

find clusters
```{r}
grp_n_pre <- find.clusters(chin_genlit_n_pre, max.n.clust=40, n.pca=100)
```

check Kstat
```{r}
grp_n_pre$Kstat
```

will use K=2
```{r}
grp_n_pre_2 <- find.clusters(chin_genlit_n_pre, max.n.clust=40, n.clust = 2)
```

run dapc
```{r}
dapc_n_pre_2 <- dapc(chin_genlit_n_pre, grp_n_pre_2$grp, n.pca = 40, n.da = 10) #keep first 40 PCs; and 10 n.da
dapc_n_pre_2
```

plot DAPC using ggplot
```{r}
dapc_n_pre_2_scores <- as.data.frame(dapc_n_pre_2$tab)
dapc_n_pre_2_scores$Assigned_Pop <- dapc_n_pre_2$grp
dapc_n_pre_2_scores$Sampling_Location <- chin_n_pre_meta$Location

set.seed(9)
dapc_n_pre_2 <- ggplot(dapc_n_pre_2_scores, aes(x=PC1, y=PC2, shape = Assigned_Pop)) + 
  geom_point(size=2, aes(colour=Sampling_Location )) + 
  stat_ellipse(level = 0.95, size = 1) + 
  scale_color_manual(values = mycol[c(1,3,4)]) + 
  geom_hline(yintercept = 0) + 
  geom_vline(xintercept = 0) + 
  theme_bw()
dapc_n_pre_2
```



### post-dam removal neutral loci

find clusters
```{r}
grp_n_post <- find.clusters(chin_genlit_n_post, max.n.clust=40, n.pca=100)
```

check Kstat
```{r}
grp_n_post$Kstat
```

will use K=2
```{r}
grp_n_post_2 <- find.clusters(chin_genlit_n_post, max.n.clust=40, n.clust = 2)
```

run dapc
```{r}
dapc_n_post_2 <- dapc(chin_genlit_n_post, grp_n_post_2$grp, n.pca = 40, n.da = 10) #keep first 40 PCs; and 10 n.da
dapc_n_post_2
```

plot DAPC using ggplot
```{r}
dapc_n_post_2_scores <- as.data.frame(dapc_n_post_2$tab)
dapc_n_post_2_scores$Assigned_Pop <- dapc_n_post_2$grp
dapc_n_post_2_scores$Sampling_Location <- chin_n_post_meta$Location

set.seed(9)
dapc_n_post_2 <- ggplot(dapc_n_post_2_scores, aes(x=PC1, y=PC2, shape = Assigned_Pop)) + 
  geom_point(size=2, aes(colour=Sampling_Location )) + 
  stat_ellipse(level = 0.95, size = 1) + 
  scale_color_manual(values = mycol[c(1,3,4)]) + 
  geom_hline(yintercept = 0) + 
  geom_vline(xintercept = 0) + 
  theme_bw()
dapc_n_post_2
```


### pre-dam removal adaptive loci

find clusters
```{r}
grp_a_pre <- find.clusters(chin_genlit_a_pre, max.n.clust=40, n.pca=100)
```

check Kstat
```{r}
grp_a_pre$Kstat
```

will use K=3
```{r}
grp_a_pre_3 <- find.clusters(chin_genlit_a_pre, max.n.clust=40, n.clust = 3)
```

run dapc
```{r}
dapc_a_pre_3 <- dapc(chin_genlit_a_pre, grp_a_pre_3$grp, n.pca = 40, n.da = 10) #keep first 40 PCs; and 10 n.da
dapc_a_pre_3
```

plot DAPC using ggplot
```{r}
dapc_a_pre_3_scores <- as.data.frame(dapc_a_pre_3$tab)
dapc_a_pre_3_scores$Assigned_Pop <- dapc_a_pre_3$grp
dapc_a_pre_3_scores$Sampling_Location <- chin_a_pre_meta$Location

set.seed(9)
dapc_a_pre_3 <- ggplot(dapc_a_pre_3_scores, aes(x=PC1, y=PC2, shape = Assigned_Pop)) + 
  geom_point(size=2, aes(colour=Sampling_Location )) + 
  stat_ellipse(level = 0.95, size = 1) + 
  scale_color_manual(values = mycol[c(1,3,4)]) + 
  geom_hline(yintercept = 0) + 
  geom_vline(xintercept = 0) + 
  theme_bw()
dapc_a_pre_3
```





### post-dam removal adaptive loci

find clusters
```{r}
grp_a_post <- find.clusters(chin_genlit_a_post, max.n.clust=40, n.pca=100)
```

check Kstat
```{r}
grp_a_post$Kstat
```

will use K=3
```{r}
grp_a_post_3 <- find.clusters(chin_genlit_a_post, max.n.clust=40, n.clust = 3)
```

run dapc
```{r}
dapc_a_post_3 <- dapc(chin_genlit_a_post, grp_a_post_3$grp, n.pca = 40, n.da = 10) #keep first 40 PCs; and 10 n.da
dapc_a_post_3
```

plot DAPC using ggplot
```{r}
dapc_a_post_3_scores <- as.data.frame(dapc_a_post_3$tab)
dapc_a_post_3_scores$Assigned_Pop <- dapc_a_post_3$grp
dapc_a_post_3_scores$Sampling_Location <- chin_a_post_meta$Location

set.seed(9)
dapc_a_post_3 <- ggplot(dapc_a_post_3_scores, aes(x=PC1, y=PC2, shape = Assigned_Pop)) + 
  geom_point(size=2, aes(colour=Sampling_Location )) + 
  stat_ellipse(level = 0.95, size = 1) + 
  scale_color_manual(values = mycol[c(1,3,4)]) + 
  geom_hline(yintercept = 0) + 
  geom_vline(xintercept = 0) + 
  theme_bw()
dapc_a_post_3
```
