---
title: "steelhead_outliertest"
author: "Kimberly Ledger"
date: "4/8/2022"
output: github_document
---

### load libraries
```{r, message=FALSE}
library(vcfR) #this package is used to visualize and manipulate VCF files
library(adegenet) #this package is used for analysis of genetic/genomic data 
library(dplyr) # data manipulation
library(tidyr) # data manipulation
```   


# Part 1: Prepare the data

## read in the steelhead (Oncorhynchus mykiss) metadata

### I changed Time == During individuals to Time == Pre in Elwha_Steelhead_Formatted_kjl.csv 
```{r}
onmy_metadata <- read.csv("~/Desktop/LG_Proj4/Elwha_datafiles/Elwha_Steelhead_Formatted_kjl.csv")
head(onmy_metadata)
nrow(onmy_metadata)
```
### create column to represent both location (AD/ID/BD/SBLR) and time (pre/post)
```{r}
onmy_metadata <- onmy_metadata %>%
  unite("Time_Location", c(Time, Location), sep= "_", remove = FALSE)
```


## read in the VCF file
```{r}
onmy_vcf <- read.vcfR("~/Desktop/LG_Proj4/Elwha_datafiles/Elwha_GTSeq_Sans_CCT.vcf")
onmy_vcf
```


# Part 2: seperate data into pre and post dam datasets and use Fst to identify outliers 

## to id outliers i will use OutFLANK 

this starts with extracting the genotypes from the vcfR object 
```{r}
geno <- extract.gt(onmy_vcf)
dim(geno)

head(geno[,1:10])
```

## check vcf file for missing data 
```{r}
myMiss_loci <- apply(geno, MARGIN = 1, function(x){ sum(is.na(x)) })  #MARGIN = 1 refers to rows (loci)
myMiss_loci <- myMiss_loci/ncol(geno)

barplot(myMiss_loci, main = "Loci")
title(ylab = "Missingness (%)")

myMiss_sample <- apply(geno, MARGIN = 2, function(x){ sum(is.na(x)) })  #MARGIN = 2 refers to columns (samples)
myMiss_sample <- myMiss_sample/nrow(geno)

barplot(myMiss_sample, main = "Samples")
title(ylab = "Missingness (%)")

hist(myMiss_sample, main = "Samples with missing data")
```

## for now just keep only samples with <20% missing data - think about changing this later
```{r}
onmy_vcf@gt <- onmy_vcf@gt[, c(TRUE, myMiss_sample < 0.2)]
onmy_vcf

geno_filtered <- extract.gt(onmy_vcf)
dim(geno_filtered)

samples <- colnames(geno_filtered)
snp <- rownames(geno_filtered)
```

after filter, we have 1324 samples... 


Notice that as our genotypes look like 0/0, 0/1, and 1/1. But OutFLANK wants them to be 0, 1, or 2. The code below fixes this problem:
```{r}
G <- geno_filtered  #we are doing this because we will be running a lot of different things with G, and if we mess up we want to be able to go back to geno

G[geno_filtered %in% c("0/0")] <- 0
G[geno_filtered  %in% c("0/1")] <- 1
G[geno_filtered %in% c("1/1")] <- 2
G[is.na(G)] <- 9
tG <- t(G)
dim(tG)
```

## seperate filtered genotype data into a pre-dam and post-dam datasets 
```{r}
pre_id <- onmy_metadata$Sample_ID[which(onmy_metadata$Time=="Pre")]
post_id <- onmy_metadata$Sample_ID[which(onmy_metadata$Time=="Post")]

geno_pre <- tG[rownames(tG) %in% pre_id,]
dim(geno_pre)
geno_post <- tG[rownames(tG) %in% post_id,]
dim(geno_post)
```

**note**: there are a few individuals with missing metadata so not all samples get included here... 


now i need to match the population metadata to the genotypes  
```{r}
samples_df <- data.frame(samples)
colnames(samples_df) <- "Sample_ID"

samples_pre <- samples_df %>%
  left_join(onmy_metadata, by = "Sample_ID") %>%
  filter(Time == "Pre")

samples_post <- samples_df %>%
  left_join(onmy_metadata, by = "Sample_ID") %>%
  filter(Time == "Post")
```


```{r}
samples_pre %>%
  group_by(Location) %>%
  summarize(n())

samples_pre %>%
  group_by(Location, Sampling_Site) %>%
  summarize(n())
```


```{r}
samples_post %>%
  group_by(Location) %>%
  summarize(n())

samples_post %>%
  group_by(Location, Sampling_Site) %>%
  summarize(n())
```

Now tG should be in the input format OutFLANK needs, with SNPs as columns and individuals as rows. 

Now we can calculate Fst for each SNP: locusNames= names our loci 1,2,3 etc popNames= names our populations with the “Location” or "Sampling_Site" labels

## first use pre dam samples 

## use "Sampling_site" as population 
```{r}
library(OutFLANK)
fst_pre <- MakeDiploidFSTMat(geno_pre,locusNames=colnames(geno_pre),popNames=samples_pre$Sampling_Site)
head(fst_pre)
```

Fst and He calcuated for each locus 

```{r}
hist(fst_pre$FST, breaks = 50)
```

Once we’ve calculated Fst between the populations for each SNP individually, we want to determine whether some SNPs are statistical outliers - that is, more differentiated than we would expect. OutFLANK does this by fitting a Chi-Squared distribution to the data and looking to see if the tails of the Chi-Squared distribution have more SNPs than expected:  

### Before running OutFLANK, check for loci with low sample sizes or unusual values of uncorrected Fst 

```{r}
plot(fst_pre$FST, fst_pre$FSTNoCorr, 
     xlim=c(-0.01,0.2), ylim=c(-0.01,0.2),
     pch=20)
abline(0,1)
```

i believe everything looks good here... no major outliers but some samples do appear a bit off

### also check for that the Fst distribution looks chi-squared distributed 

```{r}
plot(fst_pre$He, fst_pre$FSTNoCorr, pch=20, col="grey")
```

no large Fst values associated with loci of low heterozygosity... i think we are good to go 

```{r}
hist(fst_pre$FSTNoCorr, breaks=seq(0,0.4, by=0.001))
hist(fst_pre$FSTNoCorr[fst_pre$He>0.1], breaks=seq(0,0.4, by=0.001))
```

it doesn't look like much changes when i remove samples with <0.1 heterozygosity... 


details on OutFLANK can be found here: http://rstudio-pubs-static.s3.amazonaws.com/305384_9aee1c1046394fb9bd8e449453d72847.html


```{r}
OF_pre <- OutFLANK(fst_pre,LeftTrimFraction=0.05,RightTrimFraction=0.1,
         Hmin=0.1,NumberOfSamples=9,qthreshold=0.05)
## check paramaters... these are default except for a RightTrimFraction of 0.1
## the RightTrimFraction determines how many of the highest Fst values are removed before estimating th shape of the fst distribution through likelihood.  when there are potentially a large number of loci affected by spatially heterogeneous selection, it can be worth trying a higher RightTrimFraction. 
## the lowest RightTrimFraction that runs is 0.07 and there are 18 outliers
## RightTrimFraction of 0.1 has 19 outliers 
## RightTrimFraction of 0.15 has 18 outliers 
## WILL GO WITH RightTrimFraction = 0.1, because it will allow me to be more conservative in what i am considering neutral loci 
## NumberOfSamples= number of space/time populations - tried many values here and the result never changed


OutFLANKResultsPlotter(OF_pre,withOutliers=T,
                       NoCorr=T,Hmin=0.1,binwidth=0.005,
                       Zoom=F,RightZoomFraction=0.05,titletext=NULL)
```

which SNPs are statistical outliers?
```{r}
P1_pre <- pOutlierFinderChiSqNoCorr(fst_pre,Fstbar=OF_pre$FSTNoCorrbar,
                                dfInferred=OF_pre$dfInferred,qthreshold=0.05,Hmin=0.1)  

outliers_pre <- P1_pre$OutlierFlag==TRUE #which of the SNPs are outliers?
table(outliers_pre)

outlier_table_pre <- P1_pre %>%
  filter(OutlierFlag == "TRUE")
head(outlier_table_pre)
```

ONE OUTLIERS IN PRE DAM DATASET... 

Now we can make a manhattan plot! We can even plot the outliers in a different color:
```{r}
P1_pre$num <- c(1:334)
plot(P1_pre$num,P1_pre$FST,xlab="Position",ylab="FST",col=rgb(0,0,0,alpha=0.1))
points(P1_pre$num[outliers_pre],P1_pre$FST[outliers_pre],col="magenta")
```


## post dam samples 

## use "Sampling_site" as population 
```{r}
library(OutFLANK)
fst_post <- MakeDiploidFSTMat(geno_post,locusNames=colnames(geno_post),popNames=samples_post$Sampling_Site)
head(fst_post)
```

Fst and He calcuated for each locus 

```{r}
hist(fst_post$FST, breaks = 50)
```

Once we’ve calculated Fst between the populations for each SNP individually, we want to determine whether some SNPs are statistical outliers - that is, more differentiated than we would expect. OutFLANK does this by fitting a Chi-Squared distribution to the data and looking to see if the tails of the Chi-Squared distribution have more SNPs than expected:  

### Before running OutFLANK, check for loci with low sample sizes or unusual values of uncorrected Fst 

```{r}
plot(fst_post$FST, fst_post$FSTNoCorr, 
     xlim=c(-0.01,0.2), ylim=c(-0.01,0.2),
     pch=20)
abline(0,1)
```

i believe everything looks good here... no major outliers but some samples do appear a bit off

### also check for that the Fst distribution looks chi-squared distributed 

```{r}
plot(fst_post$He, fst_post$FSTNoCorr, pch=20, col="grey")
```

thre is one large Fst value associated with loci of low heterozygosity 

```{r}
hist(fst_post$FSTNoCorr, breaks=seq(0,1.0, by=0.001))
hist(fst_post$FSTNoCorr[fst_post$He>0.1], breaks=seq(0,0.8, by=0.001))
```


details on OutFLANK can be found here: http://rstudio-pubs-static.s3.amazonaws.com/305384_9aee1c1046394fb9bd8e449453d72847.html


```{r}
OF_post <- OutFLANK(fst_post,LeftTrimFraction=0.05,RightTrimFraction=0.1,
         Hmin=0.1,NumberOfSamples=9,qthreshold=0.05)
## check paramaters... these are default except for a RightTrimFraction of 0.1
## the RightTrimFraction determines how many of the highest Fst values are removed before estimating th shape of the fst distribution through likelihood.  when there are potentially a large number of loci affected by spatially heterogeneous selection, it can be worth trying a higher RightTrimFraction. 
## the lowest RightTrimFraction that runs is 0.07 and there are 18 outliers
## RightTrimFraction of 0.1 has 19 outliers 
## RightTrimFraction of 0.15 has 18 outliers 
## WILL GO WITH RightTrimFraction = 0.1, because it will allow me to be more conservative in what i am considering neutral loci 
## NumberOfSamples= number of space/time populations - tried many values here and the result never changed


OutFLANKResultsPlotter(OF_post,withOutliers=T,
                       NoCorr=T,Hmin=0.1,binwidth=0.005,
                       Zoom=F,RightZoomFraction=0.05,titletext=NULL)
```

which SNPs are statistical outliers?
```{r}
P1_post <- pOutlierFinderChiSqNoCorr(fst_post,Fstbar=OF_post$FSTNoCorrbar,
                                dfInferred=OF_post$dfInferred,qthreshold=0.05,Hmin=0.1)  

outliers_post <- P1_post$OutlierFlag==TRUE #which of the SNPs are outliers?
table(outliers_post)

outlier_table_post <- P1_post %>%
  filter(OutlierFlag == "TRUE")
outlier_table_post

```

26 OUTLIERS IN POST DAM DATASET!!! 


Now we can make a manhattan plot! We can even plot the outliers in a different color:
```{r}
P1_post$num <- c(1:334)
plot(P1_post$num,P1_post$FST,xlab="Position",ylab="FST",col=rgb(0,0,0,alpha=0.1))
points(P1_post$num[outliers_post],P1_post$FST[outliers_post],col="magenta")
```

## compare Fst outliers (postdam) to GTseq panel w/ putitively neutral/adaptive designation 

## read in loci metadata
```{r}
onmy_loci_meta <- read.csv("~/Desktop/LG_Proj4/Elwha_datafiles/Steelhead_Locus_Key_kjl.csv")
```

## read in SNP coordinate data 
```{r}
onmy_snp_coord <- read.csv("~/Desktop/LG_Proj4/Elwha_datafiles/SNP_Coordinates_CRITFC.csv")
```

join loci metadata and SNP coords 
```{r}
onmy_locus_join <- onmy_snp_coord %>%
  left_join(onmy_loci_meta, by = "Locus")
```

join outlier info to metadata for only the loci included in the vcf
```{r}
onmy_snp_pre <- onmy_locus_join %>%
  rename(LocusName = SNP) %>%
  right_join(P1_pre)

onmy_snp_post <- onmy_locus_join %>%
  rename(LocusName = SNP) %>%
  right_join(P1_post)
```

filter for only outliers 
```{r}
onmy_snp_pre_outliers <- onmy_snp_pre %>%
  filter(OutlierFlag == TRUE)
onmy_snp_pre_outliers

onmy_snp_post_outliers <- onmy_snp_post %>%
  filter(OutlierFlag == TRUE)
onmy_snp_post_outliers

onmy_snp_outliers <- rbind(onmy_snp_pre_outliers, onmy_snp_post_outliers)
onmy_snp_outliers

#write.csv(onmy_snp_outliers, "outputs/fst_outlier_loci_post_onmy.csv")
```

there are now 28 rows because the SNP NC_035104.1_11667915 has duplicate loci metadata.  


filter vct gt to remove the outlier loci 
```{r}
remove1 <- onmy_snp_outliers$LocusName

geno_nonoutlier <- geno_filtered[!rownames(geno_filtered) %in% remove1, ]
dim(geno_nonoutlier)
```


## transpose nonoutlier genotypes and combine with sample metadata 
```{r}
tG_nonout <- t(geno_nonoutlier)
tG_nonout <- as.data.frame(tG_nonout)
tG_nonout$Sample_ID <- rownames(tG_nonout)

onmy_nonout_df <- tG_nonout %>%
  left_join(onmy_metadata, by = "Sample_ID")
```

## check for deviation from HWE 

## create genind objects from pre and post dam samples 
```{r}
onmy_nonout_df_pre <- onmy_nonout_df %>%
  filter(Time == "Pre")

onmy_nonout_df_post <- onmy_nonout_df %>%
  filter(Time == "Post")

geno_n_pre <- onmy_nonout_df_pre[,c(1:307)]
rownames(geno_n_pre) <- onmy_nonout_df_pre$Sample_ID
col_geno_n_pre <- gsub("\\.", "_", colnames(geno_n_pre))
colnames(geno_n_pre) <- col_geno_n_pre

meta_pre <- onmy_nonout_df_pre[,-c(1:307)]

pre_loci <- colnames(geno_n_pre)
pre_ind <- rownames(geno_n_pre)
pre_location <- meta_pre$Location
#pre_rkm <- meta_pre$rkm
pre_site <- meta_pre$Sampling_Site

onmy_genind_pre_location <- df2genind(geno_n_pre,
                         sep="/",
                         ind.names=pre_ind,
                         loc.names=pre_loci, 
                         pop = pre_location,     
                         ploidy = 2)
onmy_genind_pre_location

# onmy_genind_pre_site <- df2genind(geno_n_pre,
#                          sep="/",
#                          ind.names=pre_ind,
#                          loc.names=pre_loci, 
#                          pop = pre_site,      
#                          ploidy = 2)
# onmy_genind_pre_site


geno_n_post <- onmy_nonout_df_post[,c(1:307)]
rownames(geno_n_post) <- onmy_nonout_df_post$Sample_ID
col_geno_n_post <- gsub("\\.", "_", colnames(geno_n_post))
colnames(geno_n_post) <- col_geno_n_post

meta_post <- onmy_nonout_df_post[,-c(1:307)]

post_loci <- colnames(geno_n_post)
post_ind <- rownames(geno_n_post)
post_location <- meta_post$Location
#post_rkm <- meta_post$rkm
post_site <- meta_post$Sampling_Site

onmy_genind_post_location <- df2genind(geno_n_post,
                         sep="/",
                         ind.names=post_ind,
                         loc.names=post_loci, 
                         pop = post_location,
                         ploidy = 2)
onmy_genind_post_location

# onmy_genind_post_site <- df2genind(geno_n_post,
#                          sep="/",
#                          ind.names=post_ind,
#                          loc.names=post_loci, 
#                          pop = post_site,
#                          ploidy = 2)
# onmy_genind_post_site
```


## check for deviations from hardy-weinberg equilibrium at the population level 
### PRE-DAM samples with LOCATION as popultion 
```{r, message=FALSE}
set.seed(5)
onmy_neutral_hw_test <- data.frame(sapply(seppop(onmy_genind_pre_location), 
                              function(ls) pegas::hw.test(ls, B = 1000)[,3])) # set B=0 to skip permutation test

onmy_neutral_hw_chisq <- t(data.matrix(onmy_neutral_hw_test))

Chisq.fdr <- matrix(p.adjust(onmy_neutral_hw_chisq,method="fdr"), 
                    nrow=nrow(onmy_neutral_hw_chisq))

# proportion of loci out of HWE 
alpha=0.05
Prop.loci.out.of.HWE <- data.frame(Chisq=apply(onmy_neutral_hw_chisq<alpha, 2, mean),
                                   Chisq.fdr=apply(Chisq.fdr<alpha, 2, mean))

# i will remove loci out of HWE in 3 or more of the populations 
loci_out_of_HWE_pre <- Prop.loci.out.of.HWE %>%
  filter(Chisq.fdr > 0.5)
loci_out_of_HWE_pre 

# for each population, the proportion of loci out of HWE
Prop.pops.out.of.HWE <- data.frame(Chisq=apply(onmy_neutral_hw_chisq<alpha, 1, mean), 
           Chisq.fdr=apply(Chisq.fdr<alpha, 1, mean))
Prop.pops.out.of.HWE             
```


## check for deviations from hardy-weinberg equilibrium at the population level 
### POST-DAM samples with LOCATION AS POP
```{r, message=FALSE}
set.seed(5)
onmy_neutral_hw_test <- data.frame(sapply(seppop(onmy_genind_post_location), 
                              function(ls) pegas::hw.test(ls, B = 1000)[,3])) # set B=0 to skip permutation test 

onmy_neutral_hw_chisq <- t(data.matrix(onmy_neutral_hw_test))

Chisq.fdr <- matrix(p.adjust(onmy_neutral_hw_chisq,method="fdr"), 
                    nrow=nrow(onmy_neutral_hw_chisq))

# proportion of loci out of HWE 
alpha=0.05
Prop.loci.out.of.HWE <- data.frame(Chisq=apply(onmy_neutral_hw_chisq<alpha, 2, mean),
                                   Chisq.fdr=apply(Chisq.fdr<alpha, 2, mean))

# i will remove loci out of HWE in 3 or more of the populations 
loci_out_of_HWE_post <- Prop.loci.out.of.HWE %>%
  filter(Chisq.fdr > 0.5)
loci_out_of_HWE_post

# for each population, the proportion of loci out of HWE
Prop.pops.out.of.HWE <- data.frame(Chisq=apply(onmy_neutral_hw_chisq<alpha, 1, mean), 
           Chisq.fdr=apply(Chisq.fdr<alpha, 1, mean))
Prop.pops.out.of.HWE   
```

filter vct gt to remove the loci out of HWE in 3 or more populations pre and/or post dam removal
```{r}
rownames(loci_out_of_HWE_pre)
rownames(loci_out_of_HWE_post)

## need to convert second "_" back to a "." to remove 
remove2 <- c("NC_035086.1_30372084",  "NC_035092.1_16334074",  "NC_035092.1_44116456", 
"NC_035080.1_68318464",  "NC_035081.1_40927121",  "NC_035087.1_37031757",  "NC_035091.1_44094440", 
"NC_035079.1_38336651",  "NW_24", "NC_035081.1_61772669", "NC_035078.1_52384497")

geno_nonoutlier_HWE <- geno_nonoutlier[!rownames(geno_nonoutlier) %in% remove2, ]
dim(geno_nonoutlier_HWE)
```


## transpose nonoutlier genotypes and loci in HWE and combine with sample metadata 
```{r}
tG_nonout <- t(geno_nonoutlier_HWE)
tG_nonout <- as.data.frame(tG_nonout)
tG_nonout$Sample_ID <- rownames(tG_nonout)

onmy_nonout_HWE_df <- tG_nonout %>%
  left_join(onmy_metadata, by = "Sample_ID")
```

## output joined non-outlier genotypes and loci in HWE with metadata for further analyses using other scripts 
```{r}
#write.csv(onmy_nonout_HWE_df, "outputs/onmy_loci4genstr_outflank_hwe.csv")
```



look at putative adaptive non-outliers that are in HWE 
```{r}
onmy_snp_adapt_nonoutlier <- onmy_snp_outliers %>%
  filter(OutlierFlag != TRUE) %>%
  filter(Status == "Adaptive")

onmy_snp_adapt_nonoutlier_hwe <- onmy_snp_adapt_nonoutlier[!onmy_snp_adapt_nonoutlier$LocusName %in% remove2, ]
onmy_snp_adapt_nonoutlier_hwe
```

there are 57 loci in the GTSeq panel that were indicated as "adaptive" that were not identified as Fst outliers and are in HWE  

```{r}
onmy_snp_adapt_nonoutlier_hwe %>%
  group_by(chromosome) %>%
  summarise(n())
```


join measures of LD with these "adaptive" non-outlier loci to help choose which to retain (max 1 "adaptive" loci per chromosome)

```{r}
#ld <- read.table("~/Desktop/LG_Proj4/vcftool_analyses/ld.geno.ld", as.is=T, row.names=NULL, header=T)
#adapt_pos <- onmy_snp_adapt_nonoutlier$Physical_Position
```

```{r}
#ld1 <- ld[ld$POS1 %in% adapt_pos,]
#ld2 <- ld[ld$POS2 %in% adapt_pos,]

#high_ld <- ld1 %>%
#  bind_rows(ld2) %>%
#  filter(R.2 > 0.02) %>%
#  arrange(desc(R.2))
```


in this file i have chosen which "adaptive" snps to keep and which to remove from population structure analyses... this was based on removing all loci with LD>0.02 and to only keep 1 "adaptive" snp per chromosome
```{r}
onmy_snp_adapt_nonoutlier_removed <- read.csv("onmy_snp_adapt_outlier_edit.csv") %>%
  filter(remove == "yes")

onmy_snp_adapt_nonoutlier_removed

onmy_snp_adapt_nonoutlier_keep <- read.csv("onmy_snp_adapt_outlier_edit.csv") %>%
  filter(remove == "no")

onmy_snp_adapt_nonoutlier_keep 
```


## filter vcf data again to get rid of multiple adapative loci on same chromosome
```{r}
remove3 <- onmy_snp_adapt_nonoutlier_removed$LocusName

geno_nonoutlier_HWE_one_adapt <- geno_nonoutlier_HWE[!rownames(geno_nonoutlier_HWE) %in% remove3, ]
dim(geno_nonoutlier_HWE_one_adapt)
```


## transpose nonoutlier genotypes and combine with sample metadata 
```{r}
tG_nonout <- t(geno_nonoutlier_HWE_one_adapt)
tG_nonout <- as.data.frame(tG_nonout)
tG_nonout$Sample_ID <- rownames(tG_nonout)

onmy_nonoutlier_HWE_one_adapt <- tG_nonout %>%
  left_join(onmy_metadata, by = "Sample_ID")
```

## output joined non-outlier genotypes with metadata for further analyses using other scripts 
```{r}
#write.csv(onmy_nonoutlier_HWE_one_adapt, "outputs/omy_loci4genstr_outflank_hwe_oneadapt.csv")
```


